sympy/solvers/ode/single.py

```diff
@@ -53,11 +53,13 @@ class SingleODEProblem:
     eq = None  # type: Expr
     func = None  # type: AppliedUndef
     sym = None  # type: Symbol
+    order = None
 
     def __init__(self, eq, func, sym):
         self.eq = eq
         self.func = func
         self.sym = sym
+        self.order = ode_order(eq, self.func)
```

Comment:
I think it would be better to have a property like:
```
    @property
    def order(self):
        if self._order is None:
            self._order = ode_order(eq, self.func)
        return self._order
```
This way we don't have to compute the order until it is needed.

Reply:
i have changed now.

Reply:
@oscarbenjamin how `is_linear` should be implemented same as `order` (i.e as property) or as method?

Reply:
I don't think it matters much. I'm also unsure whether is linear is needed. Do any of the solvers check for linearity before matching?

Reply:
`is_linear` is required for system of odes

Reply:
We can probably use it to make matching more efficient for single ODEs though. Many of the solvers are only for linear ODEs so there can be a quick check before attempting any matching.